### PR TITLE
hpcc4j-340 DOC_VERSION fix for when user supplies version in cmake co…

### PR DESCRIPTION
**Pull request by Michael-Gardner**
…mmand

Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@rpastrana please review

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -44,6 +44,21 @@ if(NOT DOC_VERSION)
...
+	string(STRIP "${version}" version)
+	execute_process(
+	COMMAND "grep" "-m 1" "<version>" "${CMAKE_CURRENT_SOURCE_DIR}/../pom.xml"
+	COMMAND "sed" "s/[ ]*<version>[0-9]*.[0-9]*.[0-9]*-//"
+	COMMAND "sed" "s/[-][^0-9]*<\\/version>$//"
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+	OUTPUT_VARIABLE stagever
+	RESULT_VARIABLE stagever_rv
+	)
+	if("${stagever_rv}" GREATER "0")
```
Comment (rpastrana) [inline docs/CMakeLists.txt:58]: this confuses me, in effect stagever_rv can only be 0 or negative? is that somehow accounting for the '-' delimiter?

File: docs/CMakeLists.txt [hunk 2]
```diff
@@ -44,6 +44,21 @@ if(NOT DOC_VERSION)
 	endif()
 	string(STRIP "${stagever}" stagever)
     set(DOC_VERSION "${version}")
+else()
+	set(version "${DOC_VERSION}")
+	string(STRIP "${version}" version)
+	execute_process(
+	COMMAND "grep" "-m 1" "<version>" "${CMAKE_CURRENT_SOURCE_DIR}/../pom.xml"
```
Comment (rpastrana) [inline docs/CMakeLists.txt:51]: why're we grepping version in the pom?

**Review by rpastrana — COMMENTED** [review]
@Michael-Gardner a couple of questions

**Comment by Michael-Gardner** [comment]
@rpastrana tested and rebased to 7.12.x

**Pull request merged**